testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/security/securitydomain/EJBContextMultipleSDTestCase.java

```diff
@@ -0,0 +1,230 @@
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+
+import javax.naming.Context;
+import javax.naming.InitialContext;
+import javax.naming.NamingException;
+
+import java.io.StringWriter;
+import java.io.PrintWriter;
+import java.io.File;
+import java.io.IOException;
+
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.util.Properties;
+import java.util.List;
+import java.util.logging.Logger;
+
+import org.jboss.as.test.integration.ejb.security.securitydomain.ejb.Hello;
+import org.jboss.as.test.integration.ejb.security.securitydomain.ejb.Info;
+import org.jboss.as.test.integration.management.util.CLITestUtil;
+import org.jboss.as.test.integration.management.util.MgmtOperationException;
+import org.jboss.as.test.integration.management.util.ServerReload;
+import org.jboss.as.test.shared.ServerSnapshot;
+import org.junit.FixMethodOrder;
+import org.junit.runners.MethodSorters;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+
+/**
+ * @author Francesco Marchioni
+ * JIRA: https://issues.redhat.com/browse/JBEAP-17862
+ * EJBContext principal is not popped back after invoking another EJB using a different Security Domain
+ */
+
+@RunWith(Arquillian.class)
+@FixMethodOrder(MethodSorters.DEFAULT)
+@ServerSetup(EJBContextMultipleSDTestCase.EJBContextMultipleSDTestCaseServerSetup.class)
+public class EJBContextMultipleSDTestCase {
+
+    private static final String ARCHIVE_NAME = "EJBContextMultipleSDTestCase";
+    protected Logger log = Logger.getLogger(this.getClass().getSimpleName());
+
+    @Deployment
+    public static Archive<?> deploy() {
+        JavaArchive jar = ShrinkWrap.create(JavaArchive.class, ARCHIVE_NAME + ".jar");
+        jar.addAsManifestResource(Hello.class.getPackage(), "jboss-ejb3.xml", "jboss-ejb3.xml");
+        jar.addPackage(Hello.class.getPackage());
+        return jar;
+    }
+
+    @ArquillianResource
+    private InitialContext initialContext;
+
+    private static final String HOST_PORT = "localhost:8080";
+    private static final String REMOTE_HTTP = "remote+http://" + HOST_PORT;
+    private static final String CTX_FACTORY = "org.wildfly.naming.client.WildFlyInitialContextFactory";
+
+    private static final String USER = "ADMIN";
+    private static final String PWD = "redhat1!";
+
+    private static Context getInitialContext() throws NamingException {
+        Properties ejbClientProperties = new Properties();
+        ejbClientProperties.put(Context.INITIAL_CONTEXT_FACTORY, CTX_FACTORY);
+        ejbClientProperties.put(Context.PROVIDER_URL, REMOTE_HTTP);
+        ejbClientProperties.put(Context.SECURITY_PRINCIPAL, USER);
+        ejbClientProperties.put(Context.SECURITY_CREDENTIALS, PWD);
+
+        Context context = new InitialContext(ejbClientProperties);
+        return context;
+    }
+
+    @Test
+    public void firstTest() throws Exception {
```

Comment:
Another minor comment but it would be good to give these test methods names that indicate what the test is trying to accomplish.